packages/atomic/src/components/common/query-summary/query-summary-common.tsx

```diff
@@ -0,0 +1,124 @@
+import {FunctionalComponent, h} from '@stencil/core';
+import {AnyBindings} from '../interface/bindings';
+import escape from 'escape-html';
+
+interface QuerySummaryCommonProps {
+  bindings: AnyBindings;
+  enableDuration: boolean;
+  querySummaryState: {
+    hasDuration: boolean;
+    durationInSeconds: number;
+    lastResult: number;
+    firstResult: number;
+    total: number;
+    query: string;
+    isLoading: boolean;
+    hasQuery: boolean;
+    hasError: boolean;
+    firstSearchExecuted: boolean;
+    hasResults: boolean;
+  };
+  setAriaLive: (message: string) => void;
+}
+
+export const QuerySummaryCommon: FunctionalComponent<
+  QuerySummaryCommonProps
+> = (props) => {
+  const getDurationInSeconds = (count: number) =>
+    props.bindings.i18n.t('in-seconds', {count});
+
+  const renderDuration = () => {
+    const shouldShow =
+      props.enableDuration && props.querySummaryState.hasDuration;
+    return `<span class="${
+      shouldShow ? '' : 'hidden'
+    }" part="duration"> ${getDurationInSeconds(
+      props.querySummaryState.durationInSeconds
+    )}</span>`;
+  };
+
+  const wrapHighlight = (content: string) => {
+    return `<span class="font-bold" part="highlight">${content}</span>`;
+  };
+
+  const wrapQuery = (content: string) => {
+    return `<span class="font-bold" part="highlight query">${content}</span>`;
+  };
+
+  const getResultOfProps = () => {
+    const locales = props.bindings.i18n.languages as string[];
+    return {
+      count: props.querySummaryState.lastResult,
+      first: props.querySummaryState.firstResult.toLocaleString(locales),
+      last: props.querySummaryState.lastResult.toLocaleString(locales),
+      total: props.querySummaryState.total.toLocaleString(locales),
+      query: props.querySummaryState.query,
+    };
+  };
+
+  const getHighlightedResultOfProps = () => {
+    const {first, last, total, query, ...options} = getResultOfProps();
+    return {
+      ...options,
+      interpolation: {escapeValue: false},
+      first: wrapHighlight(first),
+      last: wrapHighlight(last),
+      total: wrapHighlight(total),
+      query: wrapQuery(escape(query)),
+    };
+  };
+
+  const getSummary = () => {
+    if (props.querySummaryState.isLoading) {
+      return props.bindings.i18n.t('loading-results');
+    }
+    return props.bindings.i18n.t(
+      props.querySummaryState.hasQuery
+        ? 'showing-results-of-with-query'
+        : 'showing-results-of',
+      getResultOfProps()
+    );
+  };
+
+  const renderHasResults = () => {
+    const content = props.bindings.i18n.t(
+      props.querySummaryState.hasQuery
+        ? 'showing-results-of-with-query'
+        : 'showing-results-of',
+      getHighlightedResultOfProps()
+    );
+
+    return (
+      // deepcode ignore ReactSetInnerHtml: This is not React code
+      <div
+        part="results"
+        class="overflow-hidden overflow-ellipsis"
+        innerHTML={content + renderDuration()}
```

Comment:
most of the complexity with this component seems to be related to highlighting several parts of the query summary.

Perhaps we could leverage the new `LocalizedString` JSX util instead to make it much easier to read.

Additionally, there's a lot of code in this file that is unreachable by the insight query component, perhaps we wouldn't even need the `query-summary-common` component anymore if we use the util, since it may become considerably simpler.

Reply:
That's a good idea, it does simplify the code quite a bit.

As for dropping the QuerySummaryCommon entirely, I think it's still good to keep it

Still useful for common code for params extraction, dealing with plurals, aria-live.